client/app.jsx

```diff
@@ -1,8 +1,67 @@
 import React from 'react';
+import Header from './components/header';
+import Nav from './components/nav';
 import Home from './pages/home';
+import JoinGame from './pages/join-game';
+import parseRoute from './lib/parse-route';
+import PostForm from './pages/post-form';
+import Game from './pages/game';
 
 export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      navOpen: false,
+      route: parseRoute(window.location.hash),
+      gameDetails: null
+    };
+    this.handleClickNav = this.handleClickNav.bind(this);
+    this.renderPage = this.renderPage.bind(this);
+    this.joinGame = this.joinGame.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('hashchange', () => {
+      this.setState({ route: parseRoute(window.location.hash) });
+    });
+  }
+
+  handleClickNav() {
+    if (this.state.navOpen) {
+      this.setState({ navOpen: false });
+    } else {
+      this.setState({ navOpen: true });
+    }
+  }
+
+  joinGame(gameDetails) {
+    this.setState({ gameDetails });
+  }
```

Comment:
Rather than store the game in your app, i recommend just fetching it from your Game component, similar to how we did it in `react-hash-routing`. That way, you can refresh the page and still see the same game, because that component will fetch the game by its id.